File: Work/mortgage.py
# mortgage.py
#
# Exercise 1.7
principal = 500000.0
rate = 0.05
payment = 2684.11
total_paid = 0.0
month = 0

extra_payment_start_month = 61
extra_payment_end_month = 108
extra_payment = 1000 

while principal > payment:
    if month < extra_payment_start_month or month >= extra_payment_end_month:
        principal = principal * (1+rate/12) - payment
        total_paid = total_paid + payment
    else:
        principal = principal * (1+rate/12) - payment - extra_payment
        total_paid = total_paid + payment + extra_payment 
    month += 1
    # print(month, round(total_paid, ndigits=2), round(principal, ndigits=2))
    print(f'{month:^15} {round(total_paid, ndigits=2):^15} {round(principal, ndigits=2):^15}')

payment = principal
total_paid = total_paid + payment
month += 1
# print(month, round(total_paid, ndigits=2), round(principal, ndigits=2))
print(f'{month:^15} {round(total_paid, ndigits=2):^15} {round(principal, ndigits=2):^15}')

File: Work/mortgage.c
#include <stdio.h>

typedef struct 
{
    float principal;
    float rate;
    float payment;
    float total_paid;
    int month;
    int extra_payment_start_month;
    int extra_payment_end_month;
    int extra_payment;
} Mortgage;

void calculate(Mortgage mortgage);

Mortgage mortgage = {500000.0, 0.05, 2684.11, 0.0, 0, 61, 108, 1000};

int main(void)
{
    calculate(mortgage);
    return 0;
}

void calculate(Mortgage mortgage)
{
    while (mortgage.principal > mortgage.payment)
    {
        if (mortgage.month < mortgage.extra_payment_start_month || mortgage.month >= mortgage.extra_payment_end_month)
        {
            mortgage.principal = mortgage.principal * (1 + mortgage.rate / 12) - mortgage.payment;
            mortgage.total_paid += mortgage.payment;
        }
        else if (mortgage.month >= mortgage.extra_payment_start_month && 
                 mortgage.month < mortgage.extra_payment_end_month)
        {
            mortgage.principal = mortgage.principal * (1 + mortgage.rate / 12) - mortgage.payment - mortgage.extra_payment;
            mortgage.total_paid += mortgage.payment + mortgage.extra_payment;
        }
        mortgage.month++;
        printf("Month %d: Principal = %.2f, Total Paid = %.2f\n", 
                mortgage.month, mortgage.principal, mortgage.total_paid);
    }

    float last_payment = mortgage.principal * (1 + mortgage.rate / 12);
    mortgage.total_paid += last_payment; 
    mortgage.principal = 0.0; // Mortgage is paid off
    mortgage.month++;
    printf("Month %d: Final Payment = %.2f, Total Paid = %.2f\n", 
           mortgage.month, last_payment, mortgage.total_paid);
    return;
}